bundles/AdminBundle/Controller/Admin/SettingsController.php

```diff
@@ -1764,34 +1764,36 @@ public function websiteSettingsAction(Request $request)
             if ($request->get('xaction') == 'destroy') {
                 $id = $data['id'];
                 $setting = WebsiteSetting::getById($id);
-                $setting->delete();
-
-                return $this->adminJson(['success' => true, 'data' => []]);
+                if ($setting instanceof WebsiteSetting) {
+                    $setting->delete();
+                    return $this->adminJson(['success' => true, 'data' => []]);
+                }
             } elseif ($request->get('xaction') == 'update') {
                 // save routes
                 $setting = WebsiteSetting::getById($data['id']);
-
-                switch ($setting->getType()) {
-                    case 'document':
-                    case 'asset':
-                    case 'object':
-                        if (isset($data['data'])) {
-                            $path = $data['data'];
-                            $element = null;
-                            if ($path != null) {
-                                $element = Element\Service::getElementByPath($setting->getType(), $path);
+                if ($setting instanceof WebsiteSetting) {
+                    switch ($setting->getType()) {
+                        case 'document':
+                        case 'asset':
+                        case 'object':
+                            if (isset($data['data'])) {
+                                $path = $data['data'];
+                                $element = null;
+                                if ($path != null) {
```

Comment:
This `if` should not be required, example:
```php
$data = [];
$data['data'] = null;
var_dump(isset($data['data'])); // bool(false)
```

Reply:
Good point. Done!